src/v/raft/consensus.cc

```diff
@@ -1826,12 +1826,7 @@ consensus::do_write_snapshot(model::offset last_included_index, iobuf&& data) {
           _last_snapshot_index = last_included_index;
           _last_snapshot_term = term;
           // update configuration manager
-          return _configuration_manager
-            .add(_last_snapshot_index, std::move(cfg))
```

Comment:
can you clarify--perhaps with an example--the commit message. it says

```
 Inserting configuration at arbitrary offset in the
manager ...
```

but looking at this code it doesn't seem like configuration is being inserted at an _arbitrary_ location. rather above we 

`auto config = _configuration_manager.get(last_included_index);`

and then here `.add(_last_snapshot_index, std::move(cfg))` which sort of looks like it duplicates the entry and then truncates right before it?

how do we arrive at the point where `all the offsets following the one that was inserted will change.` ?

Reply:
adjusted the commit message